openquake/hazardlib/gsim/base.py

```diff
@@ -1009,3 +1012,502 @@ def __getitem__(self, imt):
             (co, (min_above[co] - max_below[co]) * ratio + max_below[co])
             for co in max_below
         )
+
+def hdf_arrays_to_dict(hdfgroup):
+    """
+    Convert an hdf5 group containins only data sets to a dictionary of
+    data sets
+    :param hdfgroup:
+        Instance of :class: h5py.Group
+    :returns:
+        Dictionary containing each of the datasets within the group arranged
+        by name
+    """
+    return dict([(key, hdfgroup[key][:]) for key in hdfgroup.keys()])
+
+
+class AmplificationTable(object):
+    """
+    Class to apply amplification from the GMPE tables.
+
+    :param shape:
```

Comment:
Here and below you want to use

:attr shape:

etc, since those of the attributes of the AmplificationTable, not the parameters passed at instantiation time